aten/src/ATen/native/native_functions.yaml

```diff
@@ -991,7 +991,9 @@
   dispatch:
     QuantizedCPU: empty_affine_quantized_cpu
 
-- func: _empty_per_channel_affine_quantized_like(Tensor self, Tensor zero_points, int[] size, int[] axis, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, MemoryFormat? memory_format=contiguous_format) -> Tensor
+# it's a factory function receiving a tensor argument, thus overriding explicitly
+- func: _empty_per_channel_affine_quantized(int[] size, *, Tensor scales, Tensor zero_points, int[] axis, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, MemoryFormat? memory_format=contiguous_format) -> Tensor
```

Comment:
I think this is all a pre-existing issue with TensorOptions, but I don't see how the dispatch makes sense here.

For example, if I don't specify a dtype, I won't try to dispatch to QuantizedCPU and presumably I'll just get some crappy error message about _empty_per_channel_affine_quantized not being implemented for CPU backend.

But really...dtype is not optional.  So maybe we don't want to do anything until @izdeby breaks apart TensorOptions, but it seems reasonable to provide a CPU overload and to give a real error message here.

Reply:
Good idea, let me fix the error message.Btw, it applies to other qtensor factory functions too